File: rust/src/entities/duty.rs
use serde::{Deserialize, Serialize};

use super::{action::Action, constraint::Constraint, permission::Permission};

#[derive(Clone, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Duty {
    pub assignee: Option<String>,
    pub assigner: Option<String>,
    pub consequence: Option<Box<Duty>>,
    pub target: Option<String>,
    pub uid: Option<String>,
    pub constraints: Option<Vec<Constraint>>,
    pub parent_ermission: Option<Permission>,
    pub action: Option<Action>,
}

File: src/entities/policy/duty.ts
import { Action } from "./action";
import { Constraint } from "./constraint";
import { Permission } from "./permission";

export interface Duty {
  assignee?: string;
  assigner?: string;
  consequence?: Duty;
  target?: string;
  uid?: string;
  constraint?: Constraint[];
  parentPermission?: Permission;
  action: Action | string;
}
